hydromt/models/model_api.py

```diff
@@ -1336,7 +1385,7 @@ def write_states(self, fn="states/{name}.nc", **kwargs) -> None:
             filename relative to model root and should contain a {name} placeholder,
             by default 'states/{name}.nc'
         **kwargs:
-            Additional keyword arguments that are passed to the `RasterDatasetAdapter`
+            Additional keyword arguments that are passed to the `_write_nc`
```

Comment:
Should we refer to internal function `_write_nc` in the docstring, which itself does not have a docstring?
This method could be added to the external api (remove underscore) with docstrings. It could potentially even be moved to the io.py script. Alternately it we mention only the underlying xarray.to_netcdf method here. That's were most kwargs go to.

Reply:
Maybe it's a reflex, but if possible I'd prefer keeping things private as much as we can, so I would prefer refering to `xarray.to_netcdf` instead. @hboisgon what do you think?

Reply:
I saw that we referenced it in a lot of places already but without an API docstring. The thing is that for the equivalent _read_nc method, the new 'load' option I implemented is actually then not passed to the open_dataset **kwargs so I have to document this better anyhow. I think these functions could easily be moved to io.py if you also agree?

Reply:
I don't have strong feelings on moving it to `io.py` but if there is a way you think that the documentation could be improved, then I'm always happy with that. I'll leave it up to you though, I trust your judgement.

Reply:
I did not realize but as @savente93 implemented the nice temporary file writting and saving to defered file list, it became quite tough to move this function to ``io.py``. So in the end I decide to leave both ``read_nc`` and ``write_nc`` in model_api.py, removed the underscore and added docstrings. Should all be consistent now!